dist/dialog.linker.build.js

```diff
@@ -102,13 +113,24 @@ window.jQuery = window.jQuery || window.shoestring;
 			return;
 		}
 
-		// if someone is calling close directly and the hash for this dialog is in
-		// the url then we need to go back, this will trigger the hashchange binding
-		// in init
-		// NOTE the bindings seem better in the constructor e.g.
-		// "#foo".indexOf("foo") === 1
-		if(window.location.hash.replace(/^#/, "") === this.hash){
-			window.history.back();
+		// if close() is called directly and the hash for this dialog is at the end of the url,
+		// then we need to change the hash to remove it, either by going back if we can, or by adding a history
+		// state that doesn't have it at the end
+		if( window.location.hash.split( "#" ).pop() === this.hash ){
+			// let's check if the first segment in the hash is the same as the first segment in the initial hash
+			// if not, it's safe to use back() to close this out and clean the hash up
+			var firstHashSegment = window.location.hash.split( "#" )[ 1 ];
+			var firstInitialHashSegment = this.initialLocationHash.split( "#" )[ 1 ];
+			if( firstHashSegment && firstInitialHashSegment && firstInitialHashSegment !== firstHashSegment ){
+				window.history.back();
+			}
```

Comment:
Hmm. I don't think actively opening a dialog should ever trigger `history.back`. Opening feels like a step forward, and additive to history, even if it's a place you've been before. 

This logic is just about what to do when closing, which is destructive to history if we are able to use it that way... its just that sometimes we can't because the open dialog is the first thing in our session.